cmd/minikube/cmd/mount.go

```diff
@@ -266,3 +279,58 @@ func getPort() (int, error) {
 	defer l.Close()
 	return l.Addr().(*net.TCPAddr).Port, nil
 }
+
+// removePidFromFile looks at the default locations for the mount-pids file,
+// for the profile in use. If a file is found and its content shows PID, PID gets removed.
+func removePidFromFile(pid int) error {
+	profile := viper.GetString("profile")
+	paths := []string{
+		localpath.MiniPath(), // legacy mount-process path for backwards compatibility
+		localpath.Profile(profile),
+	}
+
+	for _, path := range paths {
+		err := removePid(path, pid)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// removePid reads the file at PATH and tries to remove PID from it if found
+func removePid(path string, pid int) error {
+	// is it the file we're looking for?
+	pidPath := filepath.Join(path, constants.MountProcessFileName)
+	if _, err := os.Stat(pidPath); os.IsNotExist(err) {
+		return nil
+	}
+
+	// we found the correct file
+	// we're reading the pids...
+	out, err := os.ReadFile(pidPath)
+	if err != nil {
+		return errors.Wrap(err, "ReadFile")
+	}
+
+	pids := []int{}
+	strPids := strings.Fields(string(out))
+	for _, p := range strPids {
+		intPid, err := strconv.Atoi(p)
+		if err != nil {
+			return errors.Wrap(err, "while converting pids")
+		}
+
+		// we skip the pid we're looking for
+		if intPid == pid {
+			continue
+		}
+
+		pids = append(pids, intPid)
+	}
+
+	// we convert the pids list back to string and write it back to file
+	newPids := fmt.Sprintf("%s ", strings.Trim(strings.Join(strings.Fields(fmt.Sprint(pids)), " "), "[]"))
```

Comment:
This line is a bit complex, currently this func takes in an int and then convert everything to an int to compare, but I think it would make everything easier if you convert the incoming pid to a string. Then `strconv.Atoi(p)` could be removed, and this could change into:

```suggestion
	newPids := strings.Join(pids, " ")
```

I'll include a couple code suggestions to hopefully make more sense of it